sebsync.py

```diff
@@ -138,42 +151,84 @@ def get_remote_ebooks() -> None:
-                    root = ElementTree.parse(file)
-                    ns = {"container": "urn:oasis:names:tc:opendocument:xmlns:container"}
-                    rootfile = root.find(".//container:rootfile", ns).attrib["full-path"]
-                with zip.open(rootfile) as file:
-                    root = ElementTree.parse(file)
-                    ns = {
-                        "opf": "http://www.idpf.org/2007/opf",
-                        "dc": "http://purl.org/dc/elements/1.1/",
-                    }
-                    metadata = root.find("opf:metadata", ns)
-                    id = metadata.find("dc:identifier", ns)
-                    if id is None or "standardebooks.org" not in id.text:
-                        continue
-                    modified = metadata.find(".//opf:meta[@property='dcterms:modified']", ns)
+    if options.type == "kindle":
+        # Retrieve the id, title, path, modified date, and hexdigest from our cache file
+        for path in options.books.glob("**/*.azw3"):
+            if not path.is_file():
+                continue
+            try:
+                if path.name in local_cache:
                     local_ebook = LocalEbook(
-                        id=id.text,
-                        title=metadata.find(".//dc:title", ns).text,
+                        id=local_cache[path.name].get("id"),
+                        title=local_cache[path.name].get("title"),
                         path=path,
-                        modified=fromisoformat(modified.text),
+                        modified=fromisoformat(local_cache[path.name].get("modified")),
+                        hexdigest=local_cache[path.name].get("hexdigest"),
                     )
                     local_ebooks.append(local_ebook)
-        except Exception:
-            echo_status(path, Status.UNKNOWN)
+                else:
+                    hexdigest = calculate_hash(path)
+                    for entry in set(local_cache):
+                        if local_cache[entry]["hexdigest"] == hexdigest:
+                            if options.verbose:
+                                click.echo(f"Re-indexing '{entry}' as '{path.name}'")
+                            if options.debug:
+                                click.echo(local_cache[entry])
+                            local_ebook = LocalEbook(
+                                id=local_cache[entry].get("id"),
+                                title=local_cache[entry].get("title"),
+                                path=path,
+                                modified=fromisoformat(local_cache[entry].get("modified")),
+                                hexdigest=local_cache[entry].get("hexdigest"),
+                            )
+                            local_ebooks.append(local_ebook)
+                            local_cache[path.name] = local_cache[entry]
+                            local_cache.pop(entry, None)
+            except Exception as e:
```

Comment:
I understand why to check for this exception in the case of EPUBs (corrupt zip file, unexpected internal files, unexpected XML structure). Unclear to me when an exception would be raised for a kinde book, especially as we never actually open it to process its content.